### PR TITLE
Added `Scp3114::Slapped` Event

### PR DESCRIPTION
**Pull request by CatyFoxt**
Added Scp3114 Slapped Event

### CODE REVIEW

**Comment by VALERA771** [comment]
We prefer to use Hurt method cuz it's more common and covering all such events. I don't really think that this is useful

**Comment by CatyFoxt** [comment]
@VALERA771, I saw that Scp939 has a Clawed event. That is why i created this event. And it is for the helpful for new developers because it is much more spesific for just Slapping. Hurt method can contains all type of hurting that Scp3114 can do.

**Comment by VALERA771** [comment]
> @VALERA771, I saw that Scp939 has a Clawed event. That is why i created this event. And it is for the helpful for new developers because it is much more spesific for just Slapping. Hurt method can contains all type of hurting that Scp3114 can do.

Clawed event is also firied when you claws "air" (aka no target). As far as I can see, your event won't work like this

**Comment by CatyFoxt** [comment]
@VALERA771, You are wrong the event i did is firing every time Scp3114 tries to Slap. This contains **air, targets and every slapping action.**

**Comment by VALERA771** [comment]
Okay, then yeah

File: Exiled.Events/Patches/Events/Scp3114/Slapped.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    /// </summary>
+    [EventPatch(typeof(Scp3114), nameof(Scp3114.Slapped))]
+    [HarmonyPatch(typeof(Scp3114Slap), nameof(Scp3114Slap.DamagePlayers))]
+    internal class Slapped
+    {
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(instructions);
+
+            newInstructions.InsertRange(0, new CodeInstruction[]
```
Comment (louis1706) [inline Exiled.Events/Patches/Events/Scp3114/Slapped.cs:33]: You call your Event Slapp**ed** but it's call before the DamagePlayer method is executed and so we cannot differentiate a player who have get it's damage or not

**Review by louis1706 — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Events/Scp3114/Slapped.cs (hunk 1 above)
Comment (louis1706) [reply Exiled.Events/Patches/Events/Scp3114/Slapped.cs:33]: also it's do not contain a lot of information

File: Exiled.Events/EventArgs/Scp3114/SlappedEventArgs.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        /// </param>
+        public SlappedEventArgs(Player player)
+        {
+            Player = player;
+            Scp3114 = Player.Role.As<Scp3114Role>();
+        }
+
+        /// <summary>
+        /// Gets the SCP-3114.
+        /// </summary>
```
Comment (VALERA771) [inline Exiled.Events/EventArgs/Scp3114/SlappedEventArgs.cs:34]:
```suggestion
        /// <inheritdoc/>
```

File: Exiled.Events/Handlers/Scp3114.cs [hunk 3]
```diff
@@ -38,6 +38,11 @@ public static class Scp3114
         /// </summary>
         public static Event<RevealedEventArgs> Revealed { get; set; } = new();
 
+        /// <summary>
+        /// Invoked when SCP-3114 slaps a player.
```
Comment (VALERA771) [inline Exiled.Events/Handlers/Scp3114.cs:42]:
```suggestion
        /// Invoked after SCP-3114 slaps a player.
```

File: Exiled.Events/Handlers/Scp3114.cs [hunk 4]
```diff
@@ -72,6 +77,12 @@ public static class Scp3114
         /// <param name="ev">The <see cref="RevealedEventArgs" /> instance.</param>
         public static void OnRevealed(RevealedEventArgs ev) => Revealed.InvokeSafely(ev);
 
+        /// <summary>
+        /// Called when slapped.
```
Comment (VALERA771) [inline Exiled.Events/Handlers/Scp3114.cs:81]:
```suggestion
        /// Called after SCP-3114 slaps a player.
```

**Review by VALERA771 — COMMENTED** [review]
Some doc suggestions to make them "more clawed"

**Comment by CatyFoxt** [comment]
@louis1706, I changed the index of the method and fixed the orders of the using lines.

**Comment by louis1706** [comment]
> @louis1706, I changed the index of the method and fixed the orders of the using lines.

i thinks this place would be better to put the code 
			SlappedEventArgs ev = new(Player.Get(Owner), _syncAttack);
			Handlers.Scp3114.OnSlapped(ev)

			float num = this._humeShield.CurValue + ev.HumeShieldReward;
			this._humeShield.CurValue = Mathf.Min(num, this._humeShield.MaxValue);

**Comment by louis1706** [comment]
you can use the const to take the HumeShield value

File: Exiled.Events/Patches/Events/Scp3114/Slapped.cs [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+                // SlappedEventArgs ev = new(player, AttackResult, target);
+                new(OpCodes.Newobj, GetDeclaredConstructors(typeof(SlappedEventArgs))[0]),
+
+                // Handlers.Scp3114.OnSlapped(ev)
+                new(OpCodes.Call, Method(typeof(Scp3114), nameof(Scp3114.OnSlapped))),
+            });
+
+            for (int z = 0; z < newInstructions.Count; z++)
+                Log.Info($"{newInstructions[z].opcode} {newInstructions[z].operand} : {string.Join(", ", newInstructions[z].labels.Select(x => newInstructions.FindIndex(y => y.operand == (object)x)))}");
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Scp3114/Slapped.cs:68]: Remove debug lines

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

**Pull request merged**